### PR TITLE
CRYOMNI3D: Fix warnings

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: engines/cryomni3d/font_manager.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void FontManager::loadFont(Common::ReadStream &font_fl) {
 	font_fl.read(font->comment, sizeof(font->comment));
 	//debug("Comment %s", font.comment);
 
-	for (uint i = 0; i < Font::kCharactersCount; i++) {
+	for (int i = 0; i < Font::kCharactersCount; i++) {
```
Comment (lephilousophe) [inline engines/cryomni3d/font_manager.cpp]: uint is good here, as it's an index. I would change kCharactersCount to be an uint as well

File: engines/cryomni3d/versailles/logic.cpp [hunk 2]
```diff
@@ -3013,23 +3013,108 @@ const char *CryOmni3DEngine_Versailles::kBombPassword = "JEMENVAISMAISLETATDEMEU
 // We use brace elision here because it's dumped from the EXE
```
Comment (lephilousophe) [inline engines/cryomni3d/versailles/logic.cpp]: As you put braces, you should remove the comment above.

File: engines/cryomni3d/versailles/menus.cpp [hunk 3]
```diff
@@ -99,9 +99,9 @@ uint CryOmni3DEngine_Versailles::displayOptions() {
 	showMouse(true);
 
 	uint hoveredBox = -1;
-	uint selectedBox;
+	uint selectedBox = 0;
```
Comment (lephilousophe) [inline engines/cryomni3d/versailles/menus.cpp]: I would put -1 on this one as 0 is a valid box.

File: engines/cryomni3d/versailles/menus.cpp [hunk 4]
```diff
@@ -99,9 +99,9 @@ uint CryOmni3DEngine_Versailles::displayOptions() {
...
 	int selectedMsg = 0;
-	uint volumeBox;
+	uint volumeBox = 0;
```
Comment (lephilousophe) [inline engines/cryomni3d/versailles/menus.cpp]: Same here

File: engines/cryomni3d/versailles/toolbar.cpp [hunk 5]
```diff
@@ -121,6 +121,8 @@ void Toolbar::updateZones() {
 		_inventoryOffset = 0;
 		_zones[10].secondary = true;
 		_zones[11].secondary = true;
+		inventoryIt = _inventory->begin();
+		inventorySelectedIt = _inventory->begin();
```
Comment (lephilousophe) [inline engines/cryomni3d/versailles/toolbar.cpp]: For these two lines, I would use _inventory->end() to ensure we point on an invalid data.
That would be a logic error to use these variables in this state

**Comment by ccawley2011** [comment]
Done.

> Just out of curiosity, which compiler issued these warnings?

This fixes all warnings with GCC 6.3.0 on Linux x86_64 and GCC 4.7.4 on RISC OS. It also fixes most warnings with Travis builds, but there's still one `declaration requires a global constructor` warning with the Mac OS X build that I'm not sure how to fix.

File: engines/cryomni3d/versailles/menus.cpp (hunk 3 above)
Comment (bluegr) [reply engines/cryomni3d/versailles/menus.cpp]: You can't put a -1 here, unless selectedBox is changed to be signed

File: engines/cryomni3d/versailles/menus.cpp (hunk 4 above)
Comment (bluegr) [reply engines/cryomni3d/versailles/menus.cpp]: Same as above

**Comment by lephilousophe** [comment]
Thanks.
Looks good. I will merge it.
For the OSX warning maybe I should enable build by default to see buildbot warnings.
I have some commits waiting for push. I will try to push all of this during the week.

**Comment by criezy** [comment]
@lephilousophe buildbot enables all engines anyway, so it already builds the cryomni3d code. However bear in mind that it is cross compiling, and thus not using the Apple version of the clang compiler.

The remaining warning is a know issue on macOS with the way the nullptr is implemented when c++11 is not enabled. Whenever you use a nullptr for a static variable you get this warning that it requires a global constructor. We have it all over the code in ScummVM, and I don't think there is much you can do (other than use `0` instead of nullptr, but this would be a step backward). The good news is that this warning goes away if you enable c++11 (with `./configure --enable-c++11`), but then instead we get hundreds, if not thousands, of warnings about missing override.

File: engines/cryomni3d/versailles/menus.cpp (hunk 3 above)
Comment (lephilousophe) [reply engines/cryomni3d/versailles/menus.cpp]: @bluegr, sorry I didn't see you commented on the issue before merging.

I did it everywhere in the code without any warning from the compiler. It sets the value to UINT_MAX.
It seems to be perfectly legal in C: https://stackoverflow.com/a/50632 so I would say it's legal in C++ too.

**Comment by lephilousophe** [comment]
@criezy It seems main problem is that XCode define nullptr although C++11 is not enabled (c++11-compat.h).
Good to know that buildbot enables every engine but it doesn't seem to compile everything each time.
I will look at the build history to see if I missed some warnings.

**Comment by criezy** [comment]
You can look at the log for the nightly builds (the ones where the revision is `???` at around 4am on https://buildbot.scummvm.org/builders/master-osx_intel). Those should be full builds I think?

File: engines/cryomni3d/versailles/menus.cpp (hunk 3 above)
Comment (criezy) [reply engines/cryomni3d/versailles/menus.cpp]: Yes, setting an unsigned int to a negative value is legal, but you might get yet another warning. ;-)

**Comment by lephilousophe** [comment]
@criezy Yep they are. And there are many warnings "warning: negative integer implicitly converted to unsigned type" :frowning: 
Maybe I should try to explicit that's unsigned using the u suffix: `-1u`?
In general how do you proceed to remove warnings on other architectures? Trial and error?

**Pull request merged**